### PR TITLE
{REST} Only extract subscription ID for ARM requests

### PR DESCRIPTION
**Pull request by jiasli**
**Description**  
Fix regression introduced in #13225.

`az rest` now only extracts subscription ID for ARM requests, not for AD Graph request.

The ID (`00000002-0000-0000-0000-000000000000`) in an AD Graph request (`https://graph.windows.net/00000002-0000-0000-0000-000000000000/applications/00000003-0000-0000-0000-000000000000?api-version=1.6`) is tenant ID. Using it as a subscription ID will cause failure in `get_raw_token`.

**Testing Guide**  
```
> az rest -u https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups?api-version=2019-07-01 --debug 1>$null
...
Retrieving token for resource https://management.core.windows.net/, subscription 0b1f6471-1bf0-4dda-aec3-cb9272f09590
```

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S171

File: src/azure-cli-core/azure/cli/core/util.py [hunk 1]
```diff
@@ -726,11 +726,12 @@ def send_raw_request(cli_ctx, method, url, headers=None, uri_parameters=None,  #
             result[key] = value
     uri_parameters = result or None
 
+    endpoints = cli_ctx.cloud.endpoints
     # If url is an ARM resource ID, like /subscriptions/xxx/resourcegroups/xxx?api-version=2019-07-01,
     # default to Azure Resource Manager.
     # https://management.azure.com/ + subscriptions/xxx/resourcegroups/xxx?api-version=2019-07-01
     if '://' not in url:
-        url = cli_ctx.cloud.endpoints.resource_manager + url.lstrip('/')
+        url = endpoints.resource_manager.rstrip('/') + url
```
Comment (MyronFanQiu) [inline src/azure-cli-core/azure/cli/core/util.py:734]: is this change necessary?

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/util.py:734]: No. Only for consistency.

**Comment by jiasli** [comment]
Add @MyronFanQiu as code owner for core as he is actively working on `send_raw_request`.
